nixos/modules/services/desktops/gnome/gnome-keyring.nix

```diff
@@ -48,6 +59,12 @@
       source = "${pkgs.gnome.gnome-keyring}/bin/gnome-keyring-daemon";
     };
 
+    environment.extraInit = lib.mkIf cfg.enableSSHSupport ''
```

Comment:
Hmm. Is there no easy way to express this with `environment.sessionVariables` or `environment.variables`?

Reply:
At first I was trying to use something like this:
```
environment.sessionVariables = {
  SSH_AUTH_SOCK = "${builtins.getEnv "XDG_RUNTIME_DIR"}/keyring/ssh";
};
```
Then I relied on what is done with SSH and GnuPG, I thought it was a better practice.

Reply:
This `${builtins.getEnv "XDG_RUNTIME_DIR"}` is `$XDG_RUNTIME_DIR` in eval, not in activation.

The current way is not that bad though:

https://github.com/NixOS/nixpkgs/blob/8535fb92661f37ff9f0da3007fbc942f7d134b41/nixos/modules/services/security/yubikey-agent.nix#L48-L52